### DISCUSSION TITLE
Trouble binding a Volume for PocketBase-Container

### DISCUSSION THREAD

**Discussion by samuscodes** [pocketbase/pocketbase#1712] [Q&A] [answered]
Hey y'all.
I'm new to PocketBase and trying to build a Docker-Image, which I'll run using docker-compose.
However, I ended up with this error:
```bash
Recreating 7e32aeea6c95_contentark-pocketbase ... error
Recreating 7e32aeea6c95_contentark-pocketbase ... 

ERROR: for 7e32aeea6c95_contentark-pocketbase  Cannot start service pocketbase: failed to create shim: OCI runtime create failed: runc create failed: unable to start container process: exec: "/pb/pocketbase": stat /pb/pocketbase: no such file or directory: unknown

ERROR: for pocketbase  Cannot start service pocketbase: failed to create shim: OCI runtime create failed: runc create failed: unable to start container process: exec: "/pb/pocketbase": stat /pb/pocketbase: no such file or directory: unknown
ERROR: Encountered errors while bringing up the project.
```

I made very sure the /pb-directory exists & contains the application, which is why I'm stumped.
Curiously, if I leave the volume out of the Dockerfile/-compose, everything runs smoothly (minus the lack of persistence).

Anyone got any idea?
Thanks in advance.

This is my Dockerfile:
```dockerfile
FROM alpine:latest

VOLUME /pb

RUN apk add --no-cache  \
    bash \
    unzip \
    ca-certificates

ARG PB_VERSION=0.11.4
# download and unzip PocketBase
ADD https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_amd64.zip /tmp/pb.zip
RUN unzip /tmp/pb.zip -d /pb/

EXPOSE 8090

# start PocketBase
CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8090"]
```

And this is my docker-compose:
```docker-compose
---
version: '3'

services:

  pocketbase:
    container_name: 'contentark-pocketbase'
    image: 'samusoft/pocketbase'
    volumes:
      - '/home/samu/IT/Projects/Pocketbase-Vol:/pb'
    ports:
      - '40001:8090'
    restart: 'unless-stopped'
    healthcheck:
      test: curl --fail http://localhost:40001/api/health || exit 1
      interval: 60s
      retries: 5
      timeout: 20s
```

**Comment by ganigeorgiev** [comment, marked as answer]
The above configuration doesn't seem correct. You are mounting a volume at the same path (`/pb`) where the binary is unzipped.

I would suggest to remove the `VOLUME /pb` from your Dockerfile and change the mounted volumes path in your docker-compose to `/pb/pb_data`, aka. `'/home/samu/IT/Projects/Pocketbase-Vol:/pb/pb_data'`.

Optionally, you can also mount `/pb/pb_migrations` if you are using the automigrate functionality OR mount a single directory that will contain both `pb_data` and `pb_migrations` but make sure that you are not overwriting the directory with the executable. You can specify custom `pb_data` and `pb_migrations` locations using the `--dir` and `--migrationsDir` flags, eg.:

```
CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8090", "--dir='/some/other/path/pb_data'", "--migrationsDir='/some/other/path/pb_migrations'"]
```

**Comment by samuscodes** [comment]
Yeah, that was my mistake. I thought the whole time pb_data was a file😅 
Also found, it's a good idea to create a /pb/pb_data-dir before the CMD-command, when building the image.

I'll leave you with my working Dockerfile:
```dockerfile
FROM alpine:latest

RUN apk add --no-cache  \
    bash \
    unzip \
    ca-certificates

ARG PB_VERSION=0.12.0
ADD https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_amd64.zip /tmp/pb.zip
RUN unzip /tmp/pb.zip -d /pb
RUN mkdir /pb/pb_data

EXPOSE 8080

CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8080"]
```
